shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java

```diff
@@ -294,6 +298,10 @@ public void onEnd(WindowInsetsAnimation animation) {
         }
       }
     }
+
+    public boolean isAnimating() {
```

Comment:
Checking the animation state combined with getVisibility() on the IME inset can be used to determine if keyboard is visible without resorting to heuristics, will implement in a separate PR.